src/renderer/components/layout/sidebar-storage.ts

```diff
@@ -0,0 +1,15 @@
+import { createStorage } from "../../utils";
+
+export interface SidebarStorageState {
+  width: number;
+  compact: boolean;
+  expanded: {
+    [itemId: string]: boolean;
+  }
```

Comment:
Better describes what we want

```suggestion
  expanded: Set<string>,
```

Reply:
Let's not postpone milestone with such micro optimisations? (could be improved later on demand / follow-up PRs)